controllers/certificaterequest/issue_certificate.go

```diff
@@ -210,3 +218,22 @@ func (r *CertificateRequestReconciler) IssueCertificate(reqLogger logr.Logger, c
 
 	return nil
 }
+
+func (r *CertificateRequestReconciler) FindZoneIDForChallenge(namespace string) (dnsZoneID string, err error) {
+	dnsZones := hivev1.DNSZoneList{}
+	err = r.Client.List(context.TODO(), &dnsZones, &client.ListOptions{Namespace: namespace})
+	if err != nil {
+		return "", err
+	}
+
+	if len(dnsZones.Items) != 1 {
+		return "", fmt.Errorf("%d dnsZone objects in a specific namespace found, expected 1 dnsZone", len(dnsZones.Items))
+	}
+
+	if *dnsZones.Items[0].Status.AWS.ZoneID != "" {
+		return filepath.Base(*dnsZones.Items[0].Status.AWS.ZoneID), nil //the format of that ID is (/hostedzone/<something>) so, we don't care about >/. Here, we only want to grab the stuff after the last /.
```

Comment:
The comments above were not yet resolved I'd either remove the else statements or use a case switch. The way I am reading this its either AWS or GCP and only one of these can happen